### DISCUSSION TITLE
How to delimit the actions of the tests

### DISCUSSION THREAD

**Discussion by kiragengis** [allure-framework/allure2#2474] [Questions & Support]
Hi,
I am using NUnit with C# and I have some tests that are performing global changes to the application. To run additional tests, I roll back those changes at the end. It looks something like this:

```
[Test]
public void Test()
{
    try
    {
        // Run the steps
    }
    catch (Exception ex)
    {
        // CaptureScreenshot
        // Insert something in the test to know that from here the steps are going to be related to the clean-up (Finally) section
    }
    finally
    {        
        // Rollback changes
    }
}
```

What I want to do is to supply a way to easily identify in the logs when the rollback steps start.

**Comment by baev** [comment]
Can't you wrap all the statements in the `finally` block in the Allure Step?

**Reply by kiragengis** [reply]
I found that using [AllureAfter] in the TearDown solves my problem. Thank you. The only challenge is that in the TearDown all the steps are displayed in Stage COMPLETED. Is that expected?
